Sources/System/Internals/WindowsSyscallAdapters.swift

```diff
@@ -105,4 +105,11 @@ internal func pwrite(
   return Int(nNumberOfBytesWritten)
 }
 
+@inline(__always)
+internal func ftruncate(
+  _ fd: Int32,
+  _ length: off_t
+) -> Int32 {
+  _chsize(fd, numericCast(length))
+}
```

Comment:
@compnerd do you have any opinions on this? Do you think that `_chsize` is the right choice for Windows or should we be using implementing this in terms of other APIs? You mentioned something about `SetFilePointerEx` and `SetEndOfFile`?

Reply:
In general, using the windows API is better.  It tends to be more flexible and integrates better by covering more cases (e.g. file descriptors are really a lookup table in the C library to the underlying kernel handle, and that may be created bypassing the registration by using the Windows API).

At the very least please use `_chsize_s` (which in fact I have a pending patch for NIO) as that uses a `__int64` for the offset rather than a `long` which is 32-bit.  It effectively is the behavioral change of `_FILE_OFFSET_BITS=64` on GNU.  I don’t see a reason for modern code to use the 32-bit offset even under register pressure on 32-bit platforms in a library.